### ISSUE TITLE
Add beforeLoad hook

### ISSUE THREAD

**Issue by waruyama** [iconfu/svg-inject#9]
Currently there is no `beforeLoad` hook, but it could be useful to change the src before sending the XHR.

Specifically it could be used to create a super bullet proof fallback that works on any browser, no matter if SVG or Javascript is supported:

```javascript
SVGInject.setOptions({
  beforeLoad: function(img) {
    img.src = img.srcset;
  }
});
```

The `img` tags must have the `srcset` attribute set like this:

```html
<img src="image.png" srcset="image.svg" width="128" height="128" onload="SVGInject(this)" />
```

All browsers without SVG support (like IE8 and older) also have no support for the `srcset` attribute, therefore only the PNG set in the `src` will be used.

All browsers with Javascript support will replace the `src` attribute with the `srcset` attribute before loading the SVG.

The only drawback is on Internet Explorer 9, 10 and 11, which will first load the PNG, and then for the injection load the SVG. This means that there is an unnecessary download of the fallback PNG on these browsers.

**Comment by dasboe** [comment]
I suggest an extended behaviour for the beforeLoad hook. The hook returns a string value which (if returned) is used instead of the src attribute value as the URL to load the SVG file.

In your use case the hook would look like this:

```
SVGInject.setOptions({
  beforeLoad: function(img) {
    return img.srcset;
  }
});
```

This provides extended flexibility, especially when not using the `onload` attribute for injection. The hook would also allow bypassing native browser loading of the SVG image:

Example:

```
<img class="inject-img" data-src="image1.svg" />
<img class="inject-img" data-src="image2.svg" />
```

```
SVGInject(document.getElementsByClassName('inject-img'), {
  beforeLoad: function(img) {
    return img.dataset.src;
  }
});
```

**Comment by dasboe** [comment]
implemented in 54dca631d9ec826fc01e036c7d0888300ba1e630
will be part of upcoming release 1.1.0
